### PR TITLE
Add invoicing params

### PR DESCRIPTION
**Pull request by jacqueline-stripe**
This PR adds and tests new invoicing related parameters:

Subscription model: billing, days_until_due
Subscription create: billing, days_until_due
Subscription update: billing, days_until_due

Invoice model: billing, due_date, number
Invoice create: billing, days_until_due, due_date
Invoice update: paid
Invoice list: subscription

### CODE REVIEW

File: invoice.go [hunk 1]
```diff
@@ -73,6 +78,9 @@ type Invoice struct {
 	Sub           string            `json:"subscription"`
 	Webhook       int64             `json:"webhooks_delivered_at"`
 	Meta          map[string]string `json:"metadata"`
+	Billing string `json:"billing"`
+	DueDate int64 `json:"due_date"`
+	Number string `json:"number"`
```
Comment (brandur-stripe) [inline invoice.go]: Can you run gofmt and commit? It's done like so: `gofmt -w *.go`.

File: invoice.go [hunk 2]
```diff
@@ -22,6 +22,10 @@ type InvoiceParams struct {
 	SubTrialEnd          int64
 	TaxPercent           float64
 	TaxPercentZero       bool
+	Billing string
```
Comment (brandur-stripe) [inline invoice.go]: Would you mind defining a separate data type for `Billing`? A little like this:
https://github.com/stripe/stripe-go/blob/master/sub.go#L5,L7

**Comment by jacqueline-stripe** [comment]
@brandur-stripe I made the fixes! These parameters are currently behind a gate so users will get an error if they use them but are not in the Beta.

File: invoice/client_test.go [hunk 3]
```diff
@@ -91,6 +96,14 @@ func TestAllInvoicesScenarios(t *testing.T) {
 		t.Errorf("Invoice tax  %v does not match expected tax 20\n", targetInvoice.Tax)
 	}
 
+	if targetInvoice.DueDate != dueDate {
+		t.Errorf("Invoice days until due %v does not match expected %v\n", targetInvoice.DueDate, dueDate)
+	}
+
+	if targetInvoice.Billing != invoiceParams.Billing {
+		t.Errorf("Invoice billing %v does not match expected send invoice\n", targetInvoice.Billing)
```
Comment (pantera-stripe) [inline invoice/client_test.go]: nit, did you mean `does not match expected send_invoice` or format the string with another`%v` and pass in  `invoiceParams.Billing`?

File: invoice/client_test.go [hunk 4]
```diff
@@ -345,6 +372,20 @@ func TestAllInvoicesScenarios(t *testing.T) {
...
+	}
+
+	targetInvoicePaid, err := Update(targetInvoiceClosed.ID, paidInvoice)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if targetInvoicePaid.Paid != true {
+		t.Errorf("Updated invoice paid status %v does not match expected true \n", targetInvoiceUpdated.Paid)
```
Comment (pantera-stripe) [inline invoice/client_test.go]: nit, extra space before the `\n`

**Comment by pantera-stripe** [comment]
Left a few mostly minor comments.

ptal @jacqueline-stripe

**Comment by pantera-stripe** [comment]
LGTM

**Comment by brandur-stripe** [comment]
@jacqueline-stripe Let us know if/when we should cut releases for these PRs.

**Comment by jacqueline-stripe** [comment]
@brandur-stripe feel free to release as early as 2/22 (We will not be launching to public until March, but similar to Multiplan Subscriptions, I believe it should be ok to release the bindings for people to test out).

**Comment by brandur-stripe** [comment]
Cool! Released as 19.6.0.

**Pull request merged**